### ISSUE TITLE
ServerCalls delays handler calling for unary streams for outdated reason

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#263]
https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/ServerCalls.java#L76

```java
// We delay calling method.invoke() until onHalfClose(), because application may call
// close(OK) inside invoke(), while close(OK) is not allowed before onHalfClose().
```

The restriction of halfClose before OK is no longer present, having been removed before open sourcing. We should update the comment at the very least. We may choose to change how it works.

**Comment by ejona86** [comment]
It seems the behavior should stay, but the comment should be updated. We still want to delay so that we can send back an error if the client sent more than one request. If we invoke the handler, then we would likely double-close the call.

**Comment by zhangkun83** [comment]
@carl-mastrangelo can you update this issue if it's already fixed?

**Comment by carl-mastrangelo** [comment]
Fixed as a side effect of 4b18f49330cfaf439b354bf0e9d74298db422fc2
